python/ray/tune/tune.py

```diff
@@ -67,6 +67,8 @@ def _report_progress(runner, reporter, done=False):
 
 def run(run_or_experiment,
         name=None,
+        metric=None,
+        mode=None,
```

Comment:
hmm - can we expose this top-level in a separate PR? This PR is getting a little large and you'll still need to string this through the schedulers to avoid inconsistencies